packages/api-server/api_server/routes/internal.py

```diff
@@ -93,13 +98,28 @@ async def process_msg(msg: Dict[str, Any], fleet_repo: FleetRepository) -> None:
         await task_repo.save_task_state(task_state)
         task_events.task_states.on_next(task_state)
 
-        if (
-            task_state.status == mdl.Status.completed
-            or task_state.status == mdl.Status.failed
-        ):
-            alert = await alert_repo.create_alert(task_state.booking.id, "task")
-            if alert is not None:
-                alert_events.alerts.on_next(alert)
+        if task_state.status == mdl.Status.completed:
+            if task_state.booking.id not in completed_tasks:
+                completed_tasks.append(task_state.booking.id)
+                alert = await alert_repo.create_alert(task_state.booking.id, "task")
+                if alert is not None:
+                    alert_events.alerts.on_next(alert)
+            else:
+                logger.warning(
+                    "Received a repeated complete task state update, please check if the fleet adapter has stalled"
+                )
+        elif task_state.status == mdl.Status.failed:
+            if task_state.booking.id not in failed_tasks:
+                failed_tasks.append(task_state.booking.id)
+
+                # TODO: check if the failure alert content is the same
+                alert = await alert_repo.create_alert(task_state.booking.id, "task")
+                if alert is not None:
+                    alert_events.alerts.on_next(alert)
+            else:
+                logger.warning(
+                    "Received a repeated failed task state update, please check if the fleet adapter has stalled"
+                )
```

Comment:
Please document the behavior of the fleet adapter when it stalls.

This workaround can cause problems, there is a chance of a false negative where a "real" alert get filtered because there are not enough alerts that pushes the booking id out of the queue. We need to use additional info to detect the duplicated alert. Time, alert id etc, alert id is ideal if available.

Reply:
I've reverted it so that we can handle it properly in a subsequent fix, most likely we'll be deprecating this push method, and just have the frontend handle the alerts.